soco/discovery.py

```diff
@@ -215,27 +243,378 @@ def any_soco():
+        )
+        try:
+            thread.start()
+        except RuntimeError:
+            # We probably can't start any more threads
+            # Cease thread creation and continue
+            _LOG.info(
+                "Runtime error starting thread number %d ... continue",
+                len(thread_list) + 1,
+            )
+            break
+        thread_list.append(thread)
+    _LOG.info("Created %d scanner threads", len(thread_list))
+
+    # Wait for all threads to finish
+    for thread in thread_list:
+        thread.join()
+    _LOG.info("All %d scanner threads terminated", len(thread_list))
+
+    # No Sonos devices found
+    if len(sonos_ip_addresses) == 0:
+        _LOG.info("No Sonos zones discovered")
+        return None
+
+    # Disable caching to prevent problems with the list of zones
+    # if there are multiple households
+    if multi_household:
+        original_cache_state = core.zone_group_state_shared_cache.enabled
+        if original_cache_state:
+            core.zone_group_state_shared_cache.enabled = False
+            _LOG.info("Disabled SoCo caching")
+
+    # Collect SoCo instances
+    zones = set()
+    for ip_address in sonos_ip_addresses:
+        if include_invisible:
+            for zone in config.SOCO_CLASS(ip_address).all_zones:
+                zones.add(zone)
+        else:
+            for zone in config.SOCO_CLASS(ip_address).visible_zones:
+                zones.add(zone)
+        # Stop after first zone unless we want exhaustively to find
+        # all zones across all households
+        if not multi_household:
+            break
+
+    # Restore the original cache state if required
+    if multi_household:
+        if original_cache_state:
+            core.zone_group_state_shared_cache.enabled = True
+            _LOG.info("Re-enabled SoCo caching")
+
+    _LOG.info(
+        "Include_invisible: %s | multi_household: %s | %d Zones: %s",
+        include_invisible,
+        multi_household,
+        len(zones),
+        zones,
+    )
+
+    return zones
+
+
+def scan_network_by_household_id(
+    household_id, include_invisible=False, **network_scan_kwargs
+):
+    """Convenience function to find the zones in a specific Sonos
+    Household.
+
+    Args:
+        household_id (str): The Sonos Household ID to search for. IDs take the
+            form 'Sonos_XXXXXXXXXXXXXXXXXXXXXXXXXX'.
+        include_invisible (bool, optional): Whether to include invisible Sonos devices
+            in the set of devices returned.
+        **network_scan_kwargs: Arguments for the `scan_network` function.
+            See its docstring for details. (Note that the argument
+            'multi_household' is forced to `True` when this function is
+            called.)
+
+    Returns:
+        set: A set of `SoCo` instances, one for each zone found, or else `None`.
+    """
+
+    # Take a copy to avoid creating side effects
+    new_kwargs = network_scan_kwargs.copy()
```

Comment:
This isn't necessary, since the dict is only created when the function is called.

Reply:
Didn't know that. Done.